Build failure: libtorrent-rasterbar-1_2_x

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

build *libtorrent-rasterbar-1_2_x*

### Can Hydra reproduce this build failure?

_No response_

### Link to Hydra build job

_No response_

### Relevant log output

```console
this derivation will be built:
  /nix/store/klsnrs5kspcjlhcyn96iixl0hm5wrgvy-libtorrent-rasterbar-1.2.19.drv
building '/nix/store/klsnrs5kspcjlhcyn96iixl0hm5wrgvy-libtorrent-rasterbar-1.2.19.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/dpiy5lfmiygn3a21ykdab5sg6asb1hy8-source
source root is source
Running phase: patchPhase
Running phase: autoreconfPhase
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I m4
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
libtoolize: copying file 'build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force -I m4
autoreconf: running: /nix/store/n4v3lpx257h8njmrz51rvv0128ydwisx-autoconf-2.72/bin/autoconf --force
configure.ac:122: warning: $as_echo is obsolete; use AS_ECHO(["message"]) instead
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
m4/ax_pthread.m4:89: AX_PTHREAD is expanded from...
configure.ac:122: the top level
autoreconf: configure.ac: not using Autoheader
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:69: installing 'build-aux/compile'
configure.ac:78: installing 'build-aux/config.guess'
configure.ac:78: installing 'build-aux/config.sub'
configure.ac:84: installing 'build-aux/install-sh'
configure.ac:84: installing 'build-aux/missing'
examples/Makefile.am: installing 'build-aux/depcomp'
parallel-tests: installing 'build-aux/test-driver'
test/Makefile.am:248: warning: variable 'test_pex_SOURCES' is defined but no program or
test/Makefile.am:248: library has 'test_pex' as canonical name (possible typo)
autoreconf: 'build-aux/install-sh' is updated
autoreconf: Leaving directory '.'
Running phase: updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.guess
Running phase: configurePhase
fixing libtool script ./build-aux/ltmain.sh
./configure
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/4fvc5fm8bszmkydng1ivrvr5cbvr1g60-bash-5.2p37/bin/sh"
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/5qfjffhrmlmi69wqbwywsiac50yk4q74-libtorrent-rasterbar-1.2.19 --bindir=/nix/store/5qfjffhrmlmi69wqbwywsiac50yk4q74-libtorrent-rasterbar-1.2.19/bin --sbindir=/nix/store/5qfjffhrmlmi69wqbwywsiac50yk4q74-libtorrent-rasterbar-1.2.19/sbin --includedir=/nix/store/mrppha0145lsa6gci0n9ad87fyd6bpvy-libtorrent-rasterbar-1.2.19-dev/include --oldincludedir=/nix/store/mrppha0145lsa6gci0n9ad87fyd6bpvy-libtorrent-rasterbar-1.2.19-dev/include --mandir=/nix/store/5qfjffhrmlmi69wqbwywsiac50yk4q74-libtorrent-rasterbar-1.2.19/share/man --infodir=/nix/store/5qfjffhrmlmi69wqbwywsiac50yk4q74-libtorrent-rasterbar-1.2.19/share/info --docdir=/nix/store/5qfjffhrmlmi69wqbwywsiac50yk4q74-libtorrent-rasterbar-1.2.19/share/doc/libtorrent-rasterbar --libdir=/nix/store/5qfjffhrmlmi69wqbwywsiac50yk4q74-libtorrent-rasterbar-1.2.19/lib --libexecdir=/nix/store/5qfjffhrmlmi69wqbwywsiac50yk4q74-libtorrent-rasterbar-1.2.19/libexec --localedir=/nix/store/5qfjffhrmlmi69wqbwywsiac50yk4q74-libtorrent-rasterbar-1.2.19/share/locale --enable-python-binding --with-libiconv=yes --with-boost=/nix/store/iaqb4164jz55nficgvdjspbgb517rmgs-boost-1.87.0-dev --with-boost-libdir=/nix/store/qb02hs33dinll16rq1asfhzalzll7cp2-boost-1.87.0/lib PYTHON_INSTALL_PARAMS=--prefix=\$\(DESTDIR\)\$\(prefix\)\ --single-version-externally-managed\ --record=installed-files.txt
checking whether make supports nested variables... yes

Building libtorrent-rasterbar 1.2.19

Checking for a C/C++ compiler to use:
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking how to run the C preprocessor... gcc -E
checking whether gcc and cc understand -c and -o together... yes
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking how to run the C++ preprocessor... g++ -E
checking whether g++ understands -c and -o together... yes

Checking system type:
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu

Initializing Automake:
checking for a BSD-compatible install... /nix/store/pw26z7msfmv291jq1qpp8bbaprs5hkfk-coreutils-9.5/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /nix/store/pw26z7msfmv291jq1qpp8bbaprs5hkfk-coreutils-9.5/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... none
checking dependency style of g++... none
checking whether to enable maintainer-specific portions of Makefiles... yes

Initializing Libtool:
checking how to print strings... printf
checking for a sed that does not truncate output... /nix/store/v93bdqf52x2r2lqijv0f7x3yhylwy957-gnused-4.9/bin/sed
checking for grep that handles long lines and -e... /nix/store/01dri6m7g4bx0kkd39pkgfpzpj3i7y86-gnugrep-3.11/bin/grep
checking for egrep... /nix/store/01dri6m7g4bx0kkd39pkgfpzpj3i7y86-gnugrep-3.11/bin/grep -E
checking for fgrep... /nix/store/01dri6m7g4bx0kkd39pkgfpzpj3i7y86-gnugrep-3.11/bin/grep -F
checking for ld used by gcc... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ranlib... ranlib
checking for archiver @FILE support... @
checking for strip... strip
checking command to parse nm output from gcc object... ok
checking for sysroot... no
checking for a working dd... /nix/store/pw26z7msfmv291jq1qpp8bbaprs5hkfk-coreutils-9.5/bin/dd
checking how to truncate binary pipes... /nix/store/pw26z7msfmv291jq1qpp8bbaprs5hkfk-coreutils-9.5/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... ld -m elf_x86_64
checking if the linker (ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate

Checking for posix thread support:
checking for egrep -e... (cached) /nix/store/01dri6m7g4bx0kkd39pkgfpzpj3i7y86-gnugrep-3.11/bin/grep -E
checking whether gcc is Clang... no
checking whether pthreads work with "-pthread" and "-lpthread"... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
Checking for visibility support:
checking for __attribute__((visibility("hidden")))... found
yes

Checking for boost libraries:
checking for boostlib >= 1.58 (105800) includes in "/nix/store/iaqb4164jz55nficgvdjspbgb517rmgs-boost-1.87.0-dev/include"... yes
checking for boostlib >= 1.58 (105800) lib path in "/nix/store/iaqb4164jz55nficgvdjspbgb517rmgs-boost-1.87.0-dev/lib/x86_64-linux-gnu"... no
checking for boostlib >= 1.58 (105800) lib path in "/nix/store/iaqb4164jz55nficgvdjspbgb517rmgs-boost-1.87.0-dev/lib64"... no
checking for boostlib >= 1.58 (105800) lib path in "/nix/store/iaqb4164jz55nficgvdjspbgb517rmgs-boost-1.87.0-dev/libx32"... no
checking for boostlib >= 1.58 (105800) lib path in "/nix/store/iaqb4164jz55nficgvdjspbgb517rmgs-boost-1.87.0-dev/lib"... yes
checking for boostlib >= 1.58 (105800)... yes
checking which c++ standard to use... ./configure: line 21973: test: default: integer expression expected
./configure: line 22339: test: default: integer expression expected
17
checking whether g++ supports C++17 features with -std=c++17... yes
checking whether the Boost::System library is available... yes
checking for exit in -lboost_system... yes

Checking for pkg-config:
checking pkg-config is at least version 0.20... yes

Checking for functions and headers:
checking for gcc option to enable large file support... none needed
checking whether ln -s works... yes
checking whether make sets $(MAKE)... (cached) yes
checking for clock_gettime... yes

Checking build options:
checking whether deprecated functions should be enabled... yes
checking whether debug build should be enabled... no
checking whether invariant check should be enabled... no
checking whether logging to disk should be enabled... yes

Checking features to be enabled:
checking whether encryption support should be enabled... yes
configure: encryption support: now checking for the OpenSSL library...
checking for pkg-config... pkg-config
checking whether compiling and linking against OpenSSL works... yes
checking whether dht support should be enabled... yes

Checking for extra build files:
checking whether example files should be built... no
checking whether test files should be built... no
checking whether python bindings should be built... yes
checking for a Python interpreter with version >= 3.6... python
checking for python... /nix/store/3i8fa331nr390ylkf1xka5ah0hvbvfbs-python3-3.11.11/bin/python
checking for python version... 3.11
checking for python platform... linux
checking for GNU default python prefix... ${prefix}
checking for GNU default python exec_prefix... ${exec_prefix}
checking for python script directory (pythondir)... ${PYTHON_PREFIX}/lib/python3.11/site-packages
checking for python extension module directory (pyexecdir)... ${PYTHON_EXEC_PREFIX}/lib/python3.11/site-packages
checking for python3.11... (cached) /nix/store/3i8fa331nr390ylkf1xka5ah0hvbvfbs-python3-3.11.11/bin/python
checking for a version of Python >= '2.1.0'... yes
checking for a version of Python >= '2.4'... yes
checking for the distutils Python package... yes
checking for Python include path... <string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
<string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
-I/nix/store/3i8fa331nr390ylkf1xka5ah0hvbvfbs-python3-3.11.11/include/python3.11
checking for Python library path... <stdin>:4: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<stdin>:4: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
<stdin>:3: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<stdin>:3: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
<stdin>:2: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<stdin>:2: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
-L/nix/store/3i8fa331nr390ylkf1xka5ah0hvbvfbs-python3-3.11.11/lib -lpython3.11
checking for Python site-packages path... <string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
/nix/store/3i8fa331nr390ylkf1xka5ah0hvbvfbs-python3-3.11.11/lib/python3.11/site-packages
checking python extra libraries... <string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
-ldl -L/nix/store/0szrc79hm06rprwd4v5lg80fwg4sn2wj-libxcrypt-4.4.36/lib -lm
checking python extra linking flags... <string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
-Xlinker -export-dynamic
checking consistency of all components of python development environment... yes
checking whether the Boost::Python library is available... yes
checking whether boost_python is the correct library... no
checking whether boost_python311 is the correct library... yes

Checking for external libraries:
checking whether to link against system libiconv... checking for ld used by gcc... ld -m elf_x86_64
checking if the linker (ld -m elf_x86_64) is GNU ld... yes
checking for shared library run path origin... done
checking for iconv... yes
checking for working iconv... yes
checking for iconv declaration... 
         extern size_t iconv (iconv_t cd, char * *inbuf, size_t *inbytesleft, char * *outbuf, size_t *outbytesleft);
yes

Generating Makefiles:
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating include/libtorrent/Makefile
config.status: creating examples/Makefile
config.status: creating test/Makefile
config.status: creating tools/Makefile
config.status: creating bindings/Makefile
config.status: creating bindings/python/Makefile
config.status: creating bindings/python/link_flags
config.status: creating bindings/python/compile_flags
config.status: creating bindings/python/compile_cmd
config.status: creating libtorrent-rasterbar.pc
config.status: executing depfiles commands
config.status: executing libtool commands

Configure script has finished system check.

   Config results:
  -=-=-=-=-=-=-=-=-

Package:
  name:                 libtorrent-rasterbar
  version:              1.2.19
  git revision:         e2d32cd44

Build environment:
  build system:         x86_64-pc-linux-gnu
  host system:          x86_64-pc-linux-gnu
  target system:        x86_64-pc-linux-gnu

Compiler and linker flags:
  CPPFlags:             -I/nix/store/iaqb4164jz55nficgvdjspbgb517rmgs-boost-1.87.0-dev/include 
  CFlags:               -pthread -g -O2 -fvisibility=hidden
  CXXFlags:             -g -O2 -ftemplate-depth=512 -Wno-format-zero-length -fvisibility=hidden -fvisibility-inlines-hidden
  LDFlags:              -L/nix/store/qb02hs33dinll16rq1asfhzalzll7cp2-boost-1.87.0/lib  -fvisibility=hidden -fvisibility-inlines-hidden
  Libs:                  -lboost_system -lpthread 
  Defs:                 -DPACKAGE_NAME=\"libtorrent-rasterbar\" -DPACKAGE_TARNAME=\"libtorrent-rasterbar\" -DPACKAGE_VERSION=\"1.2.19\" -DPACKAGE_STRING=\"libtorrent-rasterbar\ 1.2.19\" -DPACKAGE_BUGREPORT=\"arvid@libtorrent.org\" -DPACKAGE_URL=\"http://www.libtorrent.org\" -DPACKAGE=\"libtorrent-rasterbar\" -DVERSION=\"1.2.19\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_PTHREAD_PRIO_INHERIT=1 -DHAVE_PTHREAD=1 -DHAVE_BOOST=/\*\*/ -DHAVE_CXX17=1 -DHAVE_BOOST_SYSTEM=/\*\*/ -DHAVE_CLOCK_GETTIME=1 -DNDEBUG=1 -DTORRENT_USE_INVARIANT_CHECKS=0 -DTORRENT_USE_OPENSSL=1 -DTORRENT_USE_LIBCRYPTO=1 -DHAVE_PYTHON=\"3.11\" -DHAVE_BOOST_PYTHON=/\*\*/ -DHAVE_ICONV=1 -DICONV_CONST= -DBOOST_ASIO_HAS_STD_CHRONO=1 -DBOOST_EXCEPTION_DISABLE=1 -DBOOST_ASIO_ENABLE_CANCELIO=1 -DTORRENT_BUILDING_SHARED=1

Build options:
  deprecated functions: yes
  debug build:          no
  invariant checks:     no
  logging support:      yes
  cxx standard:         17

Features:
  encryption support:   yes
  dht support:          yes

Extra builds:
  examples:             no
  tests:                no
  python bindings:      yes

Pthread library:
  CFlags:               -pthread
  Libs:                 -lpthread

Boost libraries:
  version:              
  CPPFlags:             -I/nix/store/iaqb4164jz55nficgvdjspbgb517rmgs-boost-1.87.0-dev/include
  LDFlags:              -L/nix/store/qb02hs33dinll16rq1asfhzalzll7cp2-boost-1.87.0/lib
  boost.system:         -lboost_system
  boost.python:         -lboost_python311

Python environment:
  -automake-
  binary:               /nix/store/3i8fa331nr390ylkf1xka5ah0hvbvfbs-python3-3.11.11/bin/python
  version:              3.11
  platform:             linux
  prefix:               ${prefix}
  exec_prefix:          ${exec_prefix}
  pythondir:            ${PYTHON_PREFIX}/lib/python3.11/site-packages
  pkgpythondir:         ${pythondir}/libtorrent-rasterbar
  pyexecdir:            ${PYTHON_EXEC_PREFIX}/lib/python3.11/site-packages
  pkgpyexecdir:         ${pyexecdir}/libtorrent-rasterbar

  -m4-
  cppflags:             -I/nix/store/3i8fa331nr390ylkf1xka5ah0hvbvfbs-python3-3.11.11/include/python3.11
  ldflags:              
  extra libs:           -ldl -L/nix/store/0szrc79hm06rprwd4v5lg80fwg4sn2wj-libxcrypt-4.4.36/lib -lm


External libraries:
  system libiconv:      yes

Iconv library:
  Iconv Libs:           

OpenSSL library:
  OpenSSL Libs:         -lssl -lcrypto
  OpenSSL LDFlags:      -L/nix/store/l1rb9mrm7ij5pxg79j0025pc62pry1mm-openssl-3.3.2/lib
  OpenSSL Includes:     -I/nix/store/qkll7s2yqp9mgcpjc086b48byajswx2c-openssl-3.3.2-dev/include

Type 'make' to compile libtorrent-rasterbar 1.2.19
or type 'make V=1' for verbose compiling
and then 'make install' to install it into /nix/store/5qfjffhrmlmi69wqbwywsiac50yk4q74-libtorrent-rasterbar-1.2.19

Running phase: buildPhase
build flags: -j16 SHELL=/nix/store/4fvc5fm8bszmkydng1ivrvr5cbvr1g60-bash-5.2p37/bin/bash
Making all in include/libtorrent
make[1]: Entering directory '/build/source/include/libtorrent'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/build/source/include/libtorrent'
Making all in src
make[1]: Entering directory '/build/source/src'
  CXX      libtorrent_rasterbar_la-web_connection_base.lo
  CXX      libtorrent_rasterbar_la-alert.lo
  CXX      libtorrent_rasterbar_la-alert_manager.lo
  CXX      libtorrent_rasterbar_la-announce_entry.lo
  CXX      libtorrent_rasterbar_la-assert.lo
  CXX      libtorrent_rasterbar_la-bandwidth_limit.lo
  CXX      libtorrent_rasterbar_la-bandwidth_manager.lo
  CXX      libtorrent_rasterbar_la-bandwidth_queue_entry.lo
  CXX      libtorrent_rasterbar_la-bdecode.lo
  CXX      libtorrent_rasterbar_la-bitfield.lo
  CXX      libtorrent_rasterbar_la-bloom_filter.lo
  CXX      libtorrent_rasterbar_la-broadcast_socket.lo
  CXX      libtorrent_rasterbar_la-block_cache.lo
  CXX      libtorrent_rasterbar_la-bt_peer_connection.lo
  CXX      libtorrent_rasterbar_la-chained_buffer.lo
  CXX      libtorrent_rasterbar_la-choker.lo
  CXX      libtorrent_rasterbar_la-close_reason.lo
  CXX      libtorrent_rasterbar_la-cpuid.lo
  CXX      libtorrent_rasterbar_la-crc32c.lo
  CXX      libtorrent_rasterbar_la-create_torrent.lo
  CXX      libtorrent_rasterbar_la-disk_buffer_holder.lo
  CXX      libtorrent_rasterbar_la-disk_buffer_pool.lo
  CXX      libtorrent_rasterbar_la-disk_io_job.lo
  CXX      libtorrent_rasterbar_la-disk_io_thread.lo
In file included from ../include/libtorrent/disk_io_thread.hpp:40,
                 from block_cache.cpp:39:
../include/libtorrent/io_service.hpp:40:10: fatal error: boost/asio/io_service.hpp: No such file or directory
   40 | #include <boost/asio/io_service.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:1354: libtorrent_rasterbar_la-block_cache.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
In file included from ../include/libtorrent/disk_io_thread.hpp:40,
                 from create_torrent.cpp:35:
../include/libtorrent/io_service.hpp:40:10: fatal error: boost/asio/io_service.hpp: No such file or directory
   40 | #include <boost/asio/io_service.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:1403: libtorrent_rasterbar_la-create_torrent.lo] Error 1
In file included from disk_buffer_pool.cpp:37:
../include/libtorrent/io_service.hpp:40:10: fatal error: boost/asio/io_service.hpp: No such file or directory
   40 | #include <boost/asio/io_service.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:1417: libtorrent_rasterbar_la-disk_buffer_pool.lo] Error 1
In file included from ../include/libtorrent/aux_/session_interface.hpp:39,
                 from ../include/libtorrent/ip_voter.hpp:40,
                 from ../include/libtorrent/peer_list.hpp:47,
                 from ../include/libtorrent/torrent.hpp:51,
                 from choker.cpp:37:
../include/libtorrent/io_service.hpp:40:10: fatal error: boost/asio/io_service.hpp: No such file or directory
   40 | #include <boost/asio/io_service.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:1375: libtorrent_rasterbar_la-choker.lo] Error 1
In file included from ../include/libtorrent/aux_/session_interface.hpp:39,
                 from ../include/libtorrent/ip_voter.hpp:40,
                 from ../include/libtorrent/peer_list.hpp:47,
                 from ../include/libtorrent/torrent.hpp:51,
                 from ../include/libtorrent/web_connection_base.hpp:43,
                 from web_connection_base.cpp:38:
../include/libtorrent/io_service.hpp:40:10: fatal error: boost/asio/io_service.hpp: No such file or directory
   40 | #include <boost/asio/io_service.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:1270: libtorrent_rasterbar_la-web_connection_base.lo] Error 1
In file included from ../include/libtorrent/aux_/session_interface.hpp:39,
                 from ../include/libtorrent/ip_voter.hpp:40,
                 from ../include/libtorrent/peer_list.hpp:47,
                 from ../include/libtorrent/torrent.hpp:51,
                 from alert.cpp:42:
../include/libtorrent/io_service.hpp:40:10: fatal error: boost/asio/io_service.hpp: No such file or directory
   40 | #include <boost/asio/io_service.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:1277: libtorrent_rasterbar_la-alert.lo] Error 1
In file included from ../include/libtorrent/aux_/session_interface.hpp:39,
                 from ../include/libtorrent/ip_voter.hpp:40,
                 from ../include/libtorrent/peer_list.hpp:47,
                 from ../include/libtorrent/torrent.hpp:51,
                 from ../include/libtorrent/bt_peer_connection.hpp:51,
                 from bt_peer_connection.cpp:44:
../include/libtorrent/io_service.hpp:40:10: fatal error: boost/asio/io_service.hpp: No such file or directory
   40 | #include <boost/asio/io_service.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
In file included from ../include/libtorrent/disk_io_thread.hpp:40,
                 from disk_io_thread.cpp:35:
../include/libtorrent/io_service.hpp:40:10: fatal error: boost/asio/io_service.hpp: No such file or directory
   40 | #include <boost/asio/io_service.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:1361: libtorrent_rasterbar_la-bt_peer_connection.lo] Error 1
make[1]: *** [Makefile:1431: libtorrent_rasterbar_la-disk_io_thread.lo] Error 1
In file included from broadcast_socket.cpp:41:
../include/libtorrent/broadcast_socket.hpp: In member function 'boost::asio::ip::address_v4 libtorrent::broadcast_socket::socket_entry::broadcast_address() const':
../include/libtorrent/broadcast_socket.hpp:125:61: error: no matching function for call to 'boost::asio::ip::address_v4::broadcast(boost::asio::ip::address_v4, const boost::asio::ip::address_v4&) const'
  125 |                                 return address_v4::broadcast(socket->local_endpoint(ec).address().to_v4(), netmask);
      |                                        ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from include/boost/asio/ip/address.hpp:25,
                 from include/boost/asio/ip/detail/socket_option.hpp:25,
                 from include/boost/asio/ip/multicast.hpp:20,
                 from broadcast_socket.cpp:36:
include/boost/asio/ip/address_v4.hpp:249:21: note: candidate: 'static boost::asio::ip::address_v4 boost::asio::ip::address_v4::broadcast()'
include/boost/asio/ip/address_v4.hpp:249:21: note:   candidate expects 0 arguments, 2 provided
broadcast_socket.cpp: In function 'bool libtorrent::is_link_local(const boost::asio::ip::address&)':
broadcast_socket.cpp:80:39: error: 'const class boost::asio::ip::address_v4' has no member named 'to_ulong'
   80 |                 unsigned long ip = a4.to_ulong();
      |                                       ^~~~~~~~
broadcast_socket.cpp: In function 'bool libtorrent::is_local(const boost::asio::ip::address&)':
broadcast_socket.cpp:103:39: error: 'class boost::asio::ip::address_v4' has no member named 'to_ulong'
  103 |                 unsigned long ip = a4.to_ulong();
      |                                       ^~~~~~~~
broadcast_socket.cpp: In function 'bool libtorrent::is_any(const boost::asio::ip::address&)':
broadcast_socket.cpp:122:46: error: 'class boost::asio::ip::address_v6' has no member named 'to_v4'
  122 |                         return (addr.to_v6().to_v4() == address_v4::any());
      |                                              ^~~~~
make[1]: *** [Makefile:1347: libtorrent_rasterbar_la-broadcast_socket.lo] Error 1
In file included from ../include/libtorrent/torrent_handle.hpp:62,
                 from ../include/libtorrent/alert_types.hpp:38,
                 from ../include/libtorrent/alert_manager.hpp:40,
                 from alert_manager.cpp:34:
../include/libtorrent/broadcast_socket.hpp: In member function 'boost::asio::ip::address_v4 libtorrent::broadcast_socket::socket_entry::broadcast_address() const':
../include/libtorrent/broadcast_socket.hpp:125:61: error: no matching function for call to 'boost::asio::ip::address_v4::broadcast(boost::asio::ip::address_v4, const boost::asio::ip::address_v4&) const'
  125 |                                 return address_v4::broadcast(socket->local_endpoint(ec).address().to_v4(), netmask);
      |                                        ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from include/boost/asio/ip/address.hpp:25,
                 from ../include/libtorrent/address.hpp:45,
                 from ../include/libtorrent/torrent_handle.hpp:51:
include/boost/asio/ip/address_v4.hpp:249:21: note: candidate: 'static boost::asio::ip::address_v4 boost::asio::ip::address_v4::broadcast()'
include/boost/asio/ip/address_v4.hpp:249:21: note:   candidate expects 0 arguments, 2 provided
make[1]: *** [Makefile:1284: libtorrent_rasterbar_la-alert_manager.lo] Error 1
make[1]: Leaving directory '/build/source/src'
make: *** [Makefile:664: all-recursive] Error 1
error: builder for '/nix/store/klsnrs5kspcjlhcyn96iixl0hm5wrgvy-libtorrent-rasterbar-1.2.19.drv' failed with exit code 2;
       last 25 log lines:
       > broadcast_socket.cpp: In function 'bool libtorrent::is_local(const boost::asio::ip::address&)':
       > broadcast_socket.cpp:103:39: error: 'class boost::asio::ip::address_v4' has no member named 'to_ulong'
       >   103 |                 unsigned long ip = a4.to_ulong();
       >       |                                       ^~~~~~~~
       > broadcast_socket.cpp: In function 'bool libtorrent::is_any(const boost::asio::ip::address&)':
       > broadcast_socket.cpp:122:46: error: 'class boost::asio::ip::address_v6' has no member named 'to_v4'
       >   122 |                         return (addr.to_v6().to_v4() == address_v4::any());
       >       |                                              ^~~~~
       > make[1]: *** [Makefile:1347: libtorrent_rasterbar_la-broadcast_socket.lo] Error 1
       > In file included from ../include/libtorrent/torrent_handle.hpp:62,
       >                  from ../include/libtorrent/alert_types.hpp:38,
       >                  from ../include/libtorrent/alert_manager.hpp:40,
       >                  from alert_manager.cpp:34:
       > ../include/libtorrent/broadcast_socket.hpp: In member function 'boost::asio::ip::address_v4 libtorrent::broadcast_socket::socket_entry::broadcast_address() const':
       > ../include/libtorrent/broadcast_socket.hpp:125:61: error: no matching function for call to 'boost::asio::ip::address_v4::broadcast(boost::asio::ip::address_v4, const boost::asio::ip::address_v4&) const'
       >   125 |                                 return address_v4::broadcast(socket->local_endpoint(ec).address().to_v4(), netmask);
       >       |                                        ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > In file included from include/boost/asio/ip/address.hpp:25,
       >                  from ../include/libtorrent/address.hpp:45,
       >                  from ../include/libtorrent/torrent_handle.hpp:51:
       > include/boost/asio/ip/address_v4.hpp:249:21: note: candidate: 'static boost::asio::ip::address_v4 boost::asio::ip::address_v4::broadcast()'
       > include/boost/asio/ip/address_v4.hpp:249:21: note:   candidate expects 0 arguments, 2 provided
       > make[1]: *** [Makefile:1284: libtorrent_rasterbar_la-alert_manager.lo] Error 1
       > make[1]: Leaving directory '/build/source/src'
       > make: *** [Makefile:664: all-recursive] Error 1
```

### Additional context

`tribler` fails to build because of this dependency build failure.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.9-xanmod1, NixOS, 25.05 (Warbler), 25.05beta742084.9e4d5190a948`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"home-manager, nixos, nur"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Notify maintainers


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/tr/tribler/package.nix
+++ b/pkgs/by-name/tr/tribler/package.nix
@@ -8,11 +8,16 @@
   libtorrent-rasterbar-1_2_x,
   qt5,
   nix-update-script,
+  boost186,
 }:
 
 let
-  # libtorrent-rasterbar-1_2_x requires python311
-  python3 = python311;
+  # libtorrent-rasterbar-1_2_x requires python311 and boost 1.86
+  python3 = python311.override {
+    packageOverrides = final: prev: {
+      boost = boost186;
+    };
+  };
   libtorrent = (python3.pkgs.toPythonModule (libtorrent-rasterbar-1_2_x)).python;
 in
 stdenv.mkDerivation (finalAttrs: {

